api/src/routes/challenge.ts

```diff
@@ -166,5 +171,69 @@ export const challengeRoutes: FastifyPluginCallbackTypebox = (
     }
   );
 
+  fastify.post(
+    '/modern-challenge-completed',
+    {
+      schema: schemas.modernChallengeCompleted,
+      errorHandler(error, request, reply) {
+        if (error.validation) {
+          void reply.code(400);
+          return formatValidationError(error.validation);
+        } else {
+          fastify.errorHandler(error, request, reply);
+        }
+      }
+    },
+    async (req, reply) => {
+      try {
+        const { id, files, challengeType } = req.body;
+
+        const user = await fastify.prisma.user.findUniqueOrThrow({
+          where: { id: req.session.user.id }
+        });
+        const RawProgressTimestamp = user.progressTimestamps as
+          | ProgressTimestamp[]
+          | null;
+        const points = getPoints(RawProgressTimestamp);
+
+        const completedChallenge: CompletedChallenge = {
+          id,
+          files,
+          completedDate: Date.now()
+        };
+
+        if (challengeType === 14) {
```

Comment:
no need for magic number

```suggestion
        if (challengeType === challengeTypes.multifileCertProject) {
```

Reply:
Fixed it and replaced the number in the challenge test bodies I created for this test.